### ISSUE TITLE
Secrets openly shown in metric name

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#2165]
### Discussed in https://github.com/kedacore/keda/discussions/2163

<div type='discussions-op-text'>

<sup>Originally posted by **TimShilov** October  6, 2021</sup>
I just noticed that the metric name is created based on the MySQL scaler connection string (which is a secret and is stored in a Secret) which I think is not great. Is there any reason why it's not hashed?

Secrets are secrets and they should not be lying around. And Keda should not take something that is Secret and turn it into a plain text.
Basically, anyone with access to HPA can see the secret which is unexpected to me.

![Снимок экрана 2021-10-06 в 17 41 18](https://user-images.githubusercontent.com/20239045/136226735-7baac511-1ef4-4432-b3b4-c067f6575428.png)

The screenshot is taken from GKE UI. "Deployment" -> "Details". </div>
